platform-sdk/swirlds-common/src/main/java/com/swirlds/common/scratchpad/Scratchpad.java

```diff
@@ -0,0 +1,387 @@
+                }
+
+                int fieldCount = in.readInt();
+
+                for (int index = 0; index < fieldCount; index++) {
+                    final int fieldId = in.readInt();
+                    final K key = indexToFieldMap.get(fieldId);
+                    if (key == null) {
+                        throw new IOException("scratchpad file contains unknown field " + fieldId);
+                    }
+
+                    final SelfSerializable value = in.readSerializable();
+                    data.put(key, value);
+                }
+            }
+        } catch (final IOException e) {
+            throw new RuntimeException("unable to load scratchpad", e);
+        }
+    }
+
+    /**
+     * Write the data in the scratchpad to a temporary file.
+     *
+     * @return the path to the temporary file that was written
+     */
+    @NonNull
+    private Path flushToTemporaryFile() throws IOException {
+        final Path temporaryFile = buildTemporaryFile();
+        try (final SerializableDataOutputStream out = new SerializableDataOutputStream(
+                new BufferedOutputStream(new FileOutputStream(temporaryFile.toFile(), false)))) {
+
+            out.writeInt(fileVersion);
+
+            int fieldCount = 0;
+            for (final K keys : fields) {
+                if (data.get(keys) != null) {
+                    fieldCount++;
+                }
+            }
+            out.writeInt(fieldCount);
+
+            for (final K key : fields) {
+                final SelfSerializable value = data.get(key);
+                if (value != null) {
+                    out.writeInt(key.getFieldId());
+                    out.writeSerializable(value, true);
+                }
+            }
+        }
+
+        return temporaryFile;
+    }
+
+    /**
+     * Generate the path to the next scratchpad file.
+     *
+     * @return the path to the next scratchpad file
+     */
+    @NonNull
+    private Path generateNextFilePath() {
+        return scratchpadDirectory.resolve((nextScratchpadIndex++) + SCRATCHPAD_FILE_EXTENSION);
+    }
+
+    /**
+     * Get the file index of a scratchpad file.
+     *
+     * @param path the path to the scratchpad file
+     * @return the file index
+     */
+    private long getFileIndex(@NonNull final Path path) {
+        final String fileName = path.getFileName().toString();
+        return Long.parseLong(fileName.substring(0, fileName.indexOf('.')));
+    }
+
+    /**
+     * Get a list of all scratchpad files currently on disk sorted from lowest index to highest index.
+     */
+    @NonNull
+    private List<Path> getScratchpadFiles() {
+        if (!Files.exists(scratchpadDirectory)) {
+            return List.of();
+        }
+
+        final List<Path> files = new ArrayList<>();
+
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(scratchpadDirectory)) {
+            for (final var path : stream) {
```

Comment:
`final Path path : stream`

Reply:
I blame copilot! (In reality, still my fault 🙃) Fixed.